app/src/ui/repository-settings/git-ignore.tsx

```diff
@@ -3,15 +3,54 @@ import { DialogContent } from '../dialog'
 import { TextArea } from '../lib/text-area'
 import { LinkButton } from '../lib/link-button'
 import { Ref } from '../lib/ref'
+import { getGitIgnoreNames, getGitIgnoreText } from '../add-repository/gitignores'
+import { Button } from '../lib/button'
+import { IMenuItem, showContextualMenu } from '../../lib/menu-item'
+import { Octicon } from '../octicons/octicon'
+import * as OcticonSymbol from '../octicons/octicons.generated'
+import { Row } from '../lib/row'
 
 interface IGitIgnoreProps {
   readonly text: string | null
   readonly onIgnoreTextChanged: (text: string) => void
   readonly onShowExamples: () => void
 }
 
+interface IGitIgnoreState {
+  /** The names for the available gitignores. */
+  readonly gitIgnoreNames: ReadonlyArray<string> | null
+}
+
 /** A view for creating or modifying the repository's gitignore file */
-export class GitIgnore extends React.Component<IGitIgnoreProps, {}> {
+export class GitIgnore extends React.Component<IGitIgnoreProps, IGitIgnoreState> {
+  public constructor(props: IGitIgnoreProps) {
+    super(props)
+
+    this.state = {
+      gitIgnoreNames: null,
+    }
+  }
+
+  public async componentDidMount() {
+    const gitIgnoreNames = await getGitIgnoreNames()
+
+    this.setState({ gitIgnoreNames })
+  }
+
+  private onIgnoreTemplateSelected = async (templateName: string) => {
+    const templateText = await getGitIgnoreText(templateName)
+    this.props.onIgnoreTextChanged(templateText)
+  }
+
+  private onUseTemplateButtonClick = () => {
+    const gitIgnores = this.state.gitIgnoreNames || []
+    const items: IMenuItem[] = gitIgnores.map(n => ({
+      label: n,
+      action: () => this.onIgnoreTemplateSelected(n),
+    }))
+    showContextualMenu(items)
```

Comment:
A contextual menu is not the best UI way to express the list with templates. It will be better to use the `Select` component.

There is already an implementation of git ignore templates dropdown in `create-repository.tsx` the method name is `renderGitIgnores`